cloudinary_cli/modules/clone.py

```diff
@@ -0,0 +1,110 @@
+from click import command, option, style
+from cloudinary_cli.utils.utils import normalize_list_params, print_help_and_exit
+import cloudinary
+from cloudinary_cli.utils.utils import run_tasks_concurrently
+from cloudinary_cli.utils.api_utils import upload_file
+from cloudinary_cli.utils.config_utils import load_config, get_cloudinary_config, config_to_dict
+from cloudinary_cli.defaults import logger
+from cloudinary_cli.core.search import execute_single_request, handle_auto_pagination
+
+DEFAULT_MAX_RESULTS = 500
+
+
+@command("clone",
+         short_help="""Clone assets from one product environment to another.""",
+         help="""
+\b
+Clone assets from one product environment to another with/without tags and/or context (structured metadata is not currently supported).
+Source will be your `CLOUDINARY_URL` environemnt variable but you also can specify a different source using `-c/-C` option.
+Cloning restricted assets is also not supported currently.
+Format: cld clone -T <target_environment> <command options>
```

Comment:
Is the -T really necessary?  You're always going to have to specify a target environment.   If you look at the sync command, for example, local_folder and cloudinary_folder are mandatory (no option letter).